artiq/firmware/libproto_artiq/drtioaux_proto.rs

```diff
@@ -370,6 +416,47 @@ impl Packet {
                 writer.write_bool(succeeded)?;
                 writer.write_u8(retval)?;
             },
+            Packet::DmaAddTraceRequestWrite { id, last, length, trace } => {
+                writer.write_u8(0xb0)?;
+                writer.write_u32(id)?;
+                writer.write_bool(last)?;
+                // trace may be broken down to fit within drtio aux memory limit
+                // will be reconstructed by satellite
+                writer.write_bytes(&trace[0..length as usize])?;
+            },
+            Packet::DmaAddTraceReply { succeeded } => {
+                writer.write_u8(0xb1)?;
+                writer.write_bool(succeeded)?;
+            },
+            Packet::DmaRemoveTraceRequest { id } => {
+                writer.write_u8(0xb2)?;
+                writer.write_u32(id)?;
+            },
+            Packet::DmaRemoveTraceReply { succeeded } => {
+                writer.write_u8(0xb3)?;
+                writer.write_bool(succeeded)?;
+            },
+            Packet::DmaPlaybackRequest { id, timestamp } => {
+                writer.write_u8(0xb4)?;
+                writer.write_u32(id)?;
+                writer.write_u64(timestamp)?;
+            },
+            Packet::DmaPlaybackReply { succeeded } => {
+                writer.write_u8(0xb5)?;
+                writer.write_bool(succeeded)?;
+            },
+            Packet::DmaPlaybackStatus { id, error, channel, timestamp } => {
+                writer.write_u8(0xb6)?;
+                writer.write_u32(id)?;
+                writer.write_u8(error)?;
+                writer.write_u32(channel)?;
+                writer.write_u64(timestamp)?;
+            },
+            // called if Packet::DmaAddTraceRead would be written.
+            _ => {
```

Comment:
If I try to add the ``Read`` packet, Rust will complain about Vec getting copied, thus it's blanked out.
Compiler will complain about unreachable pattern if I do include this blank (and if I don't, that not all options are covered), explaining ``allow(unreachable_patterns)`` on top.